client/state/utils.js

```diff
@@ -72,9 +72,11 @@ export function isValidStateWithSchema( state, schema ) {
  *
  * @param {string} keyName name of key in action referencing item in state map
  * @param {function} reducer applied to referenced item in state map
+ * @param {function} deserializer takes entire state of collection and deserializes
+ * @param {function} serializer takes entire state of collection and serializes
  * @return {function} super-reducer applying reducer over map of keyed items
  */
-export const keyedReducer = ( keyName, reducer ) => {
+export const keyedReducer = ( keyName, reducer, { deserializer, serializer } = { deserializer: identity, serializer: identity } ) => {
```

Comment:
The reducer will return initial state, only if state is undefined.

eg
```jsx
reducer( undefined, { type: '@@calypso/INIT' } )
```

vs
```jsx
reducer( localForageState, { type: '@@calypso/INIT' } );
```